maas/resource_maas_boot_source_selection.go

```diff
@@ -0,0 +1,227 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceMAASBootSourceSelection() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a resource to manage a MAAS boot source selection.",
+		CreateContext: resourceBootSourceSelectionCreate,
+		ReadContext:   resourceBootSourceSelectionRead,
+		UpdateContext: resourceBootSourceSelectionUpdate,
+		DeleteContext: resourceBootSourceSelectionDelete,
+
+		Schema: map[string]*schema.Schema{
+			"arches": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "The architecture list for this resource",
+			},
+			"boot_source_id": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The BootSource this resource is associated with",
+			},
+			"labels": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "The label lists for this resource",
+			},
+			"os": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The Operating system for this resource",
+			},
+			"release": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The specific release of the Operating system for this resource",
+			},
+			"subarches": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "The list of subarches for this resource",
+			},
+		},
+	}
+}
+
+func resourceBootSourceSelectionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ClientConfig).Client
+
+	bootsource, err := getBootSource(client)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	bootsourceselectionParams := entity.BootSourceSelectionParams{
+		OS:        d.Get("os").(string),
+		Release:   d.Get("release").(string),
+		Arches:    convertToStringSlice(d.Get("arches").(*schema.Set).List()),
+		Subarches: convertToStringSlice(d.Get("subarches").(*schema.Set).List()),
+		Labels:    convertToStringSlice(d.Get("labels").(*schema.Set).List()),
+	}
+
+	bootsourceselection, err := client.BootSourceSelections.Create(bootsource.ID, &bootsourceselectionParams)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId((fmt.Sprintf("%v", bootsourceselection.ID)))
+
+	return resourceBootSourceSelectionRead(ctx, d, meta)
+}
+
+func resourceBootSourceSelectionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ClientConfig).Client
+
+	bootsource, err := getBootSource(client)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	bootsourceselection, err := getBootSourceSelection(client, bootsource.ID, id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
```

Comment:
Maybe we need to include this `d.SetId((fmt.Sprintf("%v", bootsourceselection.ID)))`